src/strings/ops.c

```diff
@@ -1693,13 +1718,27 @@ MVMString * MVM_string_join(MVMThreadContext *tc, MVMString *separator, MVMObjec
     }
     result->body.num_graphs = total_graphs;
 
-    /* If we just collect all the things as strands, are we within bounds, and
-     * will be come out ahead? */
-    if (total_strands < MVM_STRING_MAX_STRANDS && 16 <= total_graphs / total_strands) {
-        /* XXX TODO: Implement this, conditionalize branch thing below. */
+    /* If the separator and pieces are all strands of the same type,
+     * and there are enough strands of a large enough size, just copy
+     * them directly. */
+    if (total_strands <  MVM_STRING_MAX_STRANDS &&
+            16        <= total_graphs / total_strands &&
+            all_strands_of_same_type) {
+        MVMuint16 offset = 0;
+        result->body.storage_type    = MVM_STRING_STRAND;
+        result->body.storage.strands = allocate_strands(tc, total_strands);
+        result->body.num_strands     = total_strands;
+        for (i = 0; i < num_pieces; i++) {
+            MVMString *piece = pieces[i];
+            if (0 < i) {
```

Comment:
This is one of the places where I really don't find the "always use `<`" thing helps at all...

Reply:
Personally I agree, but copied the code from what's now the "else" branch and wanted it to be consistent. Should I change it?